### PR TITLE
Blog setup with graphQL and templates

### PR DESCRIPTION
**Pull request by slimekat**
## Motivation

Having a fully functional CMS for Lynn to write blog posts without commandline, with templates ready to style and present the markdown content.
## Changes

Added new components for making blog templates that will receive Markdown.
Setup GraphQL

## Testing

Ran gatsby develop, site displays as before
graphiQL tests allowed me to query new placeholder blogpost Markdown
## Out of Scope / Future Work

I will be continuing structure on blog templates and configuring the CMS access

### CODE REVIEW

File: src/components/post.js [hunk 1]
```diff
@@ -140,7 +140,6 @@ function Post(props) {
           turpis egestas.
         </div>{" "}
       </div>
-      <Footer />
```
Comment (lynncyrin) [inline src/components/post.js]: ⭐⭐⭐ _(must change)_ We definitely want to keep this line

File: gatsby-config.js [hunk 2]
```diff
@@ -32,5 +40,37 @@ module.exports = {
     // this (optional) plugin enables Progressive Web App + Offline functionality
     // To learn more, visit: https://gatsby.dev/offline
     // `gatsby-plugin-offline`,
+
```
Comment (lynncyrin) [inline gatsby-config.js]: ⭐ _(nice to have)_ extremely minor styling bit - you'll want to remove this empty line

File: gatsby-node.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      return Promise.reject(result.errors)
+    }
+
+    const posts = result.data.allMarkdownRemark.edges
+
+    posts.forEach(edge => {
+      const id = edge.node.id
+      createPage({
+        path: edge.node.fields.slug,
+        // tags: edge.node.frontmatter.tags,
```
Comment (lynncyrin) [inline gatsby-node.js]: ⭐⭐ _(seriously consider)_ I would dramatically prefer commented out vars come with a comment explaining why they're commented out!

File: gatsby-node.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        },
+      })
+    })
+
+  })
+}
+
+exports.onCreateNode = ({ node, actions, getNode }) => {
+  const { createNodeField } = actions
+//   fmImagesToRelative(node)  convert image paths for gatsby images
```
Comment (lynncyrin) [inline gatsby-node.js]: ❓ _(blocking question)_ not sure what this line is for?

File: src/components/Content.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+export const HTMLContent = ({ content, className }) => (
+  <div className={className} dangerouslySetInnerHTML={{ __html: content }} />
```
Comment (lynncyrin) [inline src/components/Content.js]: ⭐⭐⭐ _(must change)_ I'll need some context one why `dangerouslySetInnerHTML` is necessary here

File: static/admin/config.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    slug: "{{year}}-{{month}}-{{day}}-{{slug}}"
+    fields:
+      - {label: "Template Key", name: "templateKey", widget: "hidden", default: "story-post"}
+      - {label: "Title", name: "title", widget: "string"}
+      - {label: "Publish Date", name: "date", widget: "datetime"}
+      - {label: "Description", name: "description", widget: "text"}
+      - {label: "Featured Post", name: "featuredpost", widget: "boolean"}
+      - {label: "Featured Image", name: "featuredimage", widget: image}
+      - {label: "Body", name: "body", widget: "markdown"}
+      - {label: "Tags", name: "tags", widget: "list"}
```
Comment (lynncyrin) [inline static/admin/config.yaml]: ❓ _(blocking question)_ I'll need some info in the PR description about what this file + `src/pages/posts/post1.md` are for? 🔍

File: src/pages/index.js [hunk 7]
```diff
@@ -4,15 +4,18 @@ import Navigation from "../components/nav"
...
+        <Post />
+        <SEO title="Home" />
   </div>
-)
+  </Layout>
 
-export default IndexPage
+    )
+    
+    export default IndexPage
```
Comment (lynncyrin) [inline src/pages/index.js]: ❓ _(blocking question)_ I can't tell why there's a diff on this file?

File: src/templates/blog-post.js [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+      id
+      html
+      frontmatter {
+        date(formatString: "MMMM DD, YYYY")
+        title
+        description
+      }
+    }
+  }
+`
```
Comment (lynncyrin) [inline src/templates/blog-post.js]: ⭐⭐⭐ _(must change)_ Kate stood up a stub post component here https://github.com/lynncyrin/lynncyrin-dot-me/blob/main/src/components/post.js, I'm _pretty sure_ that's where all this code goes?

File: src/components/post.js (hunk 1 above)
Comment (slimekat) [reply src/components/post.js]: Footer is now part of the Layout

File: gatsby-config.js (hunk 2 above)
Comment (slimekat) [reply gatsby-config.js]: Will do!

File: gatsby-node.js (hunk 3 above)
Comment (slimekat) [reply gatsby-node.js]: Adding comment

File: gatsby-node.js (hunk 4 above)
Comment (slimekat) [reply gatsby-node.js]: Line 48, It's part of the more indepth image management that the netlifyCMS blog starter wants to use, I've commented it out for now, I can leave a comment about it and we can investigate if we want to go that direction later?

File: src/components/Content.js (hunk 5 above)
Comment (slimekat) [reply src/components/Content.js]: dangerouslySetInnerHTML is how most blog setups ive seen with Gatsby seem to convert from Markdown, I'm down to talk other options!

File: static/admin/config.yaml (hunk 6 above)
Comment (slimekat) [reply static/admin/config.yaml]: This is the config file for NetlifyCMS

src/pages/posts/post1.md is a test file to start working with graphQL and graphiQL to parse how the markdown needs to be structured into the templates

File: src/pages/index.js (hunk 7 above)
Comment (slimekat) [reply src/pages/index.js]: weird! I'll check that

File: src/templates/blog-post.js (hunk 8 above)
Comment (slimekat) [reply src/templates/blog-post.js]: I did not want my first commits to take over that file, as well my template is not ready yet.

File: src/pages/index.js (hunk 7 above)
Comment (slimekat) [reply src/pages/index.js]: my prettier setup seems wonk

**Pull request closed without merging**